bundles/org.openhab.binding.myuplink/src/main/java/org/openhab/binding/myuplink/internal/handler/MyUplinkGenericDeviceHandler.java

```diff
@@ -0,0 +1,286 @@
+        }
+        enqueueCommand(new GetSystems(this, this::updatePropertiesAndOnlineStatus));
+    }
+
+    /**
+     * result processor to handle online status updates
+     *
+     * @param status of command execution
+     * @param jsonObject json respone result
+     */
+    protected final void updateOnlineStatus(CommunicationStatus status, JsonObject jsonObject) {
+        String msg = Utils.getAsString(jsonObject, JSON_KEY_ERROR);
+        if (msg == null || msg.isBlank()) {
+            msg = status.getMessage();
+        }
+
+        switch (status.getHttpCode()) {
+            case OK:
+            case ACCEPTED:
+                super.updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+                break;
+            default:
+                super.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+        }
+    }
+
+    /**
+     * Disposes the thing.
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        cancelJobReference(dataPollingJobReference);
+    }
+
+    /**
+     * adds a channel.
+     */
+    @Override
+    public void registerChannel(Channel channel) {
+        ThingBuilder thingBuilder = editThing();
+        thingBuilder.withChannel(channel);
+        updateThing(thingBuilder.build());
+    }
+
+    /**
+     * will update all channels provided in the map
+     */
+    @Override
+    public void updateChannelStatus(Map<Channel, State> values) {
+        logger.debug("Handling heatpump channel update.");
+
+        for (Channel channel : values.keySet()) {
+            if (getThing().getChannels().contains(channel)) {
+                State value = values.get(channel);
+                if (value != null) {
+                    logger.debug("Channel is to be updated: {}: {}", channel.getUID().getAsString(), value);
+                    updateState(channel.getUID(), value);
+                } else {
+                    logger.debug("Value is null or not provided by myUplink Cloud (channel: {})",
+                            channel.getUID().getAsString());
+                    updateState(channel.getUID(), UnDefType.UNDEF);
+                }
+            } else {
+                logger.debug("Could not identify channel: {} for model {}", channel.getUID().getAsString(),
+                        getThing().getThingTypeUID().getAsString());
+            }
+        }
+    }
+
+    @Override
+    public void enqueueCommand(MyUplinkCommand command) {
+        MyUplinkBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            bridgeHandler.enqueueCommand(command);
+        } else {
+            // this should not happen
+            logger.warn("no bridge handler found");
+        }
+    }
+
+    private @Nullable MyUplinkBridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        return bridge == null ? null : ((MyUplinkBridgeHandler) bridge.getHandler());
+    }
```

Comment:
I am in doubt that this will work. How does the shortcut know that it has to call `getHandler` before it casts to `MyUplinkBridgeHandler`
I reverted to my version.

Reply:
applied it.